collator/src/lib.rs

```diff
@@ -334,13 +353,25 @@ impl<P, E> Worker for CollationNode<P, E> where
 			exit.clone(),
 			message_validator,
 			client.clone(),
-			task_executor,
+			task_executor.clone(),
 		));
 
-		let parachain_context = build_parachain_context.build(validation_network.clone()).unwrap();
+		let parachain_context = match build_parachain_context.build(
+			client.clone(),
+			task_executor,
+			validation_network.clone(),
+		) {
```

Comment:
maybe map_err(..)? can be used here

Reply:
Not really, as we return here a Future and not a Result.